node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,329 @@
+
+                        // Update client password with re-auth
+                        await expect(
+                            client.updateConnectionPassword(NEW_PASSWORD, true),
+                        ).resolves.toBe("OK");
+
+                        // Verify client works with new auth
+                        await client.set("test_key", "test_value");
+                        const value = await client.get("test_key");
+                        expect(value).toEqual("test_value");
+                    }, protocol);
+                },
+                TIMEOUT,
+            );
+
+            /**
+             * Test changing server password when connection is lost before password update.
+             * Verifies that the client will not be able to reach the connection under the abstraction and return an error.
+             *
+             * Some explanation for the curious mind:
+             * Our library is abstracting a connection or connections, with a lot of mechanism around it, making it behave like what we call a "client".
+             * When using standalone mode, the client is a single connection, so on disconnection the first thing it planned to do is to reconnect.
+             * Theres no reason to get other commands and to take care of them since to serve commands we need to be connected.
+             * Hence, the client will try to reconnect and will not listen try to take care of new tasks, but will let them wait in line,
+             * so the update connection password will not be able to reach the connection and will return an error.
+             * For future versions, standalone will be considered as a different animal then it is now, since standalone is not necessarily one node.
+             * It can be replicated and have a lot of nodes, and to be what we like to call "one shard cluster".
+             * So, in the future, we will have many existing connection and request can be managed also when one connection is locked,
+             *
+             */
+            it("test_update_connection_password_connection_lost_before_password_update", async () => {
+                await runTest(async (client: BaseClient) => {
+                    if (client instanceof GlideClusterClient) {
+                        return;
+                    }
+
+                    // Set a key to ensure connection is established
+                    await client.set("test_key", "test_value");
+                    // Update server password
+                    await client.configSet({ requirepass: NEW_PASSWORD });
+                    // Kill client connections
+                    await managementClient.customCommand([
+                        "CLIENT",
+                        "KILL",
+                        "TYPE",
+                        "normal",
+                    ]);
+                    // Try updating client password without immediate re-auth and with, both should fail
```

Comment:
Is that the same I reported on GH? If yes, post here links, so we can easily update the tests once fixed.